src/main/java/org/openjdk/jextract/impl/TreeMaker.java

```diff
@@ -369,10 +369,12 @@ public Declaration.Scoped createEnum(Cursor c) {
         if (c.isDefinition()) {
             List<Declaration> decls = new ArrayList<>();
             c.forEach(child -> {
-                Declaration enumConstantDecl = createTree(child);
-                if (enumConstantDecl != null) { // see CODETOOLS-7903673
-                    DeclarationString.with(enumConstantDecl, enumConstantString(c.spelling(), (Declaration.Constant) enumConstantDecl));
-                    decls.add(enumConstantDecl);
+                if (child.kind() == CursorKind.EnumConstantDecl) {
+                    Declaration enumConstantDecl = createTree(child);
+                    if (enumConstantDecl != null) {
```

Comment:
The null check here is not needed anymore. (test coverage shows the `null` branch is uncovered here as well)

Reply:
Isn't that the whole point of the fix? (e.g. skipping nulls when a "weird" cursor shows up)

Reply:
I thought the `child.kind() == CursorKind.EnumConstantDecl` supersedes that

Reply:
Ah, I missed the new check for EnumConstantDecl. I don't think we can ever return a null in there. So, yes, the extra null check is redundant.